### PR TITLE
3.1.0 Healthbot release KPIs

### PR DESCRIPTION
**Pull request by gowrisankarvs**
Following KPIs are submitted for 3.1.0 Healthbot release
1. Commit script in community_supplied
2. Interface optical related KPIs
3. Linecard optics power related KPIs
4. check-interface-status rule modified to filter interface based on description

### CODE REVIEW

File: juniper_official/Interfaces/ipv6_Addr.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
```
Comment (gvarao) [inline juniper_official/Interfaces/ipv6_Addr.yml:1]: As we discusses move IPv4 and V6 related rules into community

File: juniper_official/Interfaces/interface-throughput.playbook [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/*
+ * Playbook contains multiple rules which monitor interfaces bandwidth 
+ * utilization and notifies when anomalies are found.
+ *
```
Comment (gvarao) [inline juniper_official/Interfaces/interface-throughput.playbook:4]: s/multiple rules/a rule/g
s/when anomalies are found./anomalies when exceeds threshold./g

File: juniper_official/Interfaces/interface-throughput.playbook [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/*
+ * Playbook contains multiple rules which monitor interfaces bandwidth 
+ * utilization and notifies when anomalies are found.
+ * 
+ * 1) Rule check-interface-bandwidth-netconf,, detects the interface 
+ *    bandwidth and notifies anomalies when utilization is more than thresholds
+ */
```
Comment (gvarao) [inline juniper_official/Interfaces/interface-throughput.playbook:7]: Modify as below
* 1) Rule check-interface-bandwidth-netconf, Notifies anomalies when interface throughput exceeds threshold.

File: juniper_official/Interfaces/count-lo0-address-netconf.rule [hunk 4]
```diff
@@ -0,0 +1,138 @@
+/*
```
Comment (gvarao) [inline juniper_official/Interfaces/count-lo0-address-netconf.rule:1]: As we discussed move IPv4 and v6 loopback verifications into community

File: juniper_official/Interfaces/count-lo0-address-ipv6-netconf.rule [hunk 5]
```diff
@@ -0,0 +1,138 @@
+/*
```
Comment (gvarao) [inline juniper_official/Interfaces/count-lo0-address-ipv6-netconf.rule:1]: As we discussed move IPv4 and v6 loopback verifications into community

File: juniper_official/Interfaces/count-lo0-address-ipv6-netconf.rule [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+* 1) list-lo0-addresses,  vector is collecting the list of lo0 addresses from 
+*    the rule "check-lo0-address-netconf". This rule will set  dashboard color to green
+*     when any ipv6 address is present.Else it is set to red.
+*/
+healthbot {
+    topic interface.statistics {
+        rule count-lo0-address-ipv6-netconf {
+            synopsis "lo0 ipv6 configuration kpi";
+            description "Check whether IPV6 address is configured under lo0 interface";
+            rule-frequency 30s;
```
Comment (gvarao) [inline juniper_official/Interfaces/count-lo0-address-ipv6-netconf.rule:15]: Increase frequency to 3 mins

File: juniper_official/Interfaces/count-lo0-address-ipv6-netconf.rule [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+                type integer;
+                description "Count the IPV6 address configured in lo0 interface";
+            }
+            /*
+             * Anomaly detection logic.
+             */
+            trigger loopback-address-configuration {
+                synopsis "ipv6 loopback address";
+                description "Sets health based on if ipv6 loopback address is present";
+                frequency 90s;
```
Comment (gvarao) [inline juniper_official/Interfaces/count-lo0-address-ipv6-netconf.rule:40]: Increase frequency to 3 mins

File: juniper_official/Interfaces/count-lo0-address-netconf.rule [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+* 1) list-lo0-addresses,  vector is collecting the list of lo0 addresses from 
+*    the rule "check-lo0-address-netconf". This rule will set  dashboard color to green
+*     when any ipv4 address is present.Else it is set to red.
+*/
+healthbot {
+    topic interface.statistics {
+        rule count-lo0-address-netconf {
+            synopsis "lo0 configuration kpi";
+            description "Check whether IP address is configured under lo0 interface";
+            rule-frequency 30s;
```
Comment (gvarao) [inline juniper_official/Interfaces/count-lo0-address-netconf.rule:15]: Increase frequency to 3 mins

File: juniper_official/Interfaces/count-lo0-address-netconf.rule [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+                        status {
+                            color green;
+                            message "Loopback address is configured";
+                        }
+                    }
+                }
+            }
+            vector list-lo0-addresses {
+                path "/topic[topic-name=interface.statistics]/rule[rule-name=check-lo0-address-netconf]/lo0-address";
+                time-range 30s;
```
Comment (gvarao) [inline juniper_official/Interfaces/count-lo0-address-netconf.rule:71]: Increase frequency to 3 mins

File: juniper_official/Interfaces/check-lo0-address-netconf.rule [hunk 10]
```diff
@@ -0,0 +1,186 @@
+/*
```
Comment (gvarao) [inline juniper_official/Interfaces/check-lo0-address-netconf.rule:1]: As we discussed move IPv4 and v6 loopback verifications into community

File: juniper_official/Interfaces/check-lo0-address-netconf.rule [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+             * Otherwise the color is set to red.  
+             */
+            /*
+             * Sensor configuration to get data from network devices.
+             */
+            sensor lo0address {
+                iAgent {
+                    file InterfaceLo0Table.yml;
+                    table InterfaceLo0Table;
+                    frequency 30s;
```
Comment (gvarao) [inline juniper_official/Interfaces/check-lo0-address-netconf.rule:33]: increase frequency to 3m

File: juniper_official/Interfaces/check-lo0-address-netconf.rule [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+                type string;
+                description "Configured ip address in lo0";
+            }
+            /*
+             * Anomaly detection logic.
+             */
+            trigger loopback-address-verification {
+                synopsis "ipv4 loopback address";
+                description "Sets health based on if ipv4 loopback address is present";
+                frequency 90s;
```
Comment (gvarao) [inline juniper_official/Interfaces/check-lo0-address-netconf.rule:75]: increase frequency to 3 mins

File: juniper_official/Interfaces/check-lo0-address-ipv6-netconf.rule [hunk 13]
```diff
@@ -0,0 +1,180 @@
+/*
```
Comment (gvarao) [inline juniper_official/Interfaces/check-lo0-address-ipv6-netconf.rule:1]: 1. As we discussed move IPv4 and v6 loopback verifications into community.
2. Increase sensor and trigger frequency to 3 minutes

File: juniper_official/Interfaces/check-lacp-statistics-openconfig.rule [hunk 14]
```diff
@@ -0,0 +1,359 @@
+/*
+* Monitors lacp link status file and notifies when  anomalies are found.
```
Comment (gvarao) [inline juniper_official/Interfaces/check-lacp-statistics-openconfig.rule:2]: s/Monitors lacp link status file/Monitors lacp link status/g

File: juniper_official/Interfaces/check-ipv6-local-address-netconf.rule [hunk 15]
```diff
@@ -0,0 +1,152 @@
+/*
```
Comment (gvarao) [inline juniper_official/Interfaces/check-ipv6-local-address-netconf.rule:1]: 1. Move file to community
2. Modify sensor and trigger frequency to 3 minutes

File: juniper_official/Interfaces/InterfaceLo0TableIPV6.yml [hunk 16]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (gvarao) [inline juniper_official/Interfaces/InterfaceLo0TableIPV6.yml:1]: Move file to community

File: juniper_official/Interfaces/InterfaceLo0Table.yml [hunk 17]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (gvarao) [inline juniper_official/Interfaces/InterfaceLo0Table.yml:1]: 1. Move file to community

File: community_supplied/System/system-kpi.playbook [hunk 18]
```diff
@@ -0,0 +1,21 @@
+/*
```
Comment (gvarao) [inline community_supplied/System/system-kpi.playbook:1]: as we discussed modify playbook name to system-additional-kpis playbook

File: juniper_official/Interfaces/lacp-kpis-netconf.playbook [hunk 19]
```diff
@@ -0,0 +1,14 @@
+/*
```
[the remark this answers is not in the document]
Comment (gowrisankarvs) [reply juniper_official/Interfaces/lacp-kpis-netconf.playbook]: updated the playbook

File: juniper_official/Interfaces/interface-throughput.playbook (hunk 2 above)
Comment (gowrisankarvs) [reply juniper_official/Interfaces/interface-throughput.playbook:4]: updated the rule

File: juniper_official/Interfaces/interface-throughput.playbook (hunk 3 above)
Comment (gowrisankarvs) [reply juniper_official/Interfaces/interface-throughput.playbook:7]: updated the rule comments

File: juniper_official/Interfaces/count-lo0-address-ipv6-netconf.rule (hunk 5 above)
Comment (gowrisankarvs) [reply juniper_official/Interfaces/count-lo0-address-ipv6-netconf.rule]: moved file to community

File: juniper_official/Interfaces/count-lo0-address-ipv6-netconf.rule (hunk 6 above)
Comment (gowrisankarvs) [reply juniper_official/Interfaces/count-lo0-address-ipv6-netconf.rule]: updated the frequency

File: juniper_official/Interfaces/count-lo0-address-netconf.rule (hunk 8 above)
Comment (gowrisankarvs) [reply juniper_official/Interfaces/count-lo0-address-netconf.rule]: updated the frequency

File: juniper_official/Interfaces/count-lo0-address-netconf.rule (hunk 9 above)
Comment (gowrisankarvs) [reply juniper_official/Interfaces/count-lo0-address-netconf.rule]: updated the frequency

File: juniper_official/Interfaces/check-lo0-address-netconf.rule (hunk 10 above)
Comment (gowrisankarvs) [reply juniper_official/Interfaces/check-lo0-address-netconf.rule]: moved file to community

File: juniper_official/Interfaces/check-lo0-address-netconf.rule (hunk 11 above)
Comment (gowrisankarvs) [reply juniper_official/Interfaces/check-lo0-address-netconf.rule]: updated the frequency

File: juniper_official/Interfaces/check-lo0-address-netconf.rule (hunk 12 above)
Comment (gowrisankarvs) [reply juniper_official/Interfaces/check-lo0-address-netconf.rule]: updated the frequency

File: juniper_official/Interfaces/check-lo0-address-ipv6-netconf.rule (hunk 13 above)
Comment (gowrisankarvs) [reply juniper_official/Interfaces/check-lo0-address-ipv6-netconf.rule]: updated the rule

File: juniper_official/Interfaces/count-lo0-address-ipv6-netconf.rule (hunk 7 above)
Comment (gowrisankarvs) [reply juniper_official/Interfaces/count-lo0-address-ipv6-netconf.rule]: increased the frequency

File: juniper_official/Interfaces/count-lo0-address-netconf.rule (hunk 4 above)
Comment (gowrisankarvs) [reply juniper_official/Interfaces/count-lo0-address-netconf.rule]: moved file to community

File: juniper_official/Interfaces/ipv6_Addr.yml (hunk 1 above)
Comment (gowrisankarvs) [reply juniper_official/Interfaces/ipv6_Addr.yml]: moved file to community

File: juniper_official/Interfaces/list-total.py [hunk 20]
```diff
@@ -0,0 +1,6 @@
+from __future__ import division
```
[the remark this answers is not in the document]
Comment (gowrisankarvs) [reply juniper_official/Interfaces/list-total.py]: moved file to community

File: juniper_official/System/db-size-netconf.rule [hunk 21]
```diff
@@ -0,0 +1,207 @@
...
+* dashboard color to yellow when db usage percent
+* exceeds min threshold value. Otherwise it is set to green.
+* 
+*/
+healthbot {
+    topic system.storage {
+        rule check-db-size-netconf {
+            synopsis "db usage KPI";
+            description "To check for db usage and notify anomalies";
+            function db-usage-percent {
```
[the remark this answers is not in the document]
Comment (gowrisankarvs) [reply juniper_official/System/db-size-netconf.rule]: updated the rule

**Review by gvarao — APPROVED** [review]

**Pull request merged**